Menu/GameShell/10_Settings/Wifi/textarea.py

```diff
@@ -68,32 +71,71 @@ def AppendAndBlitText(self,alphabet):
         else:
             print("is Full %s" % "".join(self._MyWords))
 
+    def BuildBlitText(self):
+        blit_rows = [[]]
+        w = 0
+        xmargin = 5
+        endmargin = 15
+        x = self._PosX+xmargin
+        linenumber = 0
+        cursor_row = 0
+        
+        for i, v in enumerate(self._MyWords):
+            t = self._FontObj.render(v, True, (8, 135, 174))
+            t_width = t.get_width()
+            w += t_width
+            del(t)
+
+            blit_rows[linenumber].append(v)
+
+            if i == self._TextIndex - 1:
+                cursor_row = linenumber
+
+            if w + t_width >= self._Width-endmargin:
+                x = self._PosX+xmargin
+                w = 0
+                linenumber += 1
+                blit_rows.append([])
+        
+        # only paint 2 rows
+        if len(blit_rows) == 1:
+            self._BlitWords = blit_rows[0]
+            self._BlitIndex = self._TextIndex
+        elif len(blit_rows) == 2 or cursor_row < 2:
+            self._BlitWords = blit_rows[0] + blit_rows[1]
+            self._BlitIndex = self._TextIndex
+        else:
+            self._BlitWords = blit_rows[cursor_row - 1] + blit_rows[cursor_row]
+            self._BlitIndex = self._TextIndex
+            for i,v in enumerate(blit_rows):
+                if i == cursor_row - 1:
+                    break
+                self._BlitIndex -= len(v)
```

Comment:
I'm sure there's a more pythonic way to do this, but this seemed clear to read for calculating the displayed cursor index.